apps/charterafrica/src/components/Comments/DynamicLineClampedTypography.js

```diff
@@ -0,0 +1,54 @@
+import { Button, useTheme } from "@mui/material";
+import PropTypes from "prop-types";
+import React, { forwardRef, useState, useRef } from "react";
+
+import LineClampedRichTypography from "@/charterafrica/components/LineClampedRichTypography";
+
+const DynamicLineClampedTypography = forwardRef(
+  ({ comment, ...props }, ref) => {
+    const ref1 = useRef();
+    const commentHeight = ref1?.current?.offsetHeight;
+    const currentTheme = useTheme();
+    const lineHeight =
+      currentTheme.typography.p3.lineHeight *
+      currentTheme.typography.p3.fontSize;
+    const showReadMore = commentHeight >= lineHeight * 3;
+
+    const [readMoreOpen, setReadMoreOpen] = useState(false);
+    const lineClamp = !readMoreOpen ? 2 : undefined;
+
+    return (
+      <>
```

Comment:
May be this should just be a `div`/`Box`?
```suggestion
      <Box sx={sx} ref={ref}>
```